components/playgroundChatbot.tsx

```diff
@@ -358,25 +360,19 @@ export default function PlaygroundChatbot(props: {
                   Object.entries(functionJsonResponse).length === 0)
               ) {
                 if (
-                  devChatContents[idx - 1].role === "function" ||
-                  (devChatContents[idx + 1] &&
-                    devChatContents[idx + 1].role === "function")
+                  devChatContents[idx - 1]?.role === "function" ||
+                  devChatContents[idx + 1]?.role === "function"
                 ) {
                   // If the function call is adjacent to other function calls we don't need to tell them it
                   // was empty - otherwise we get a lot of empty messages clogging up the chat interface
-                  return <div key={idx + chatItem.content} />;
+                  return <div key={idx.toString() + chatItem.content} />;
                 }
                 contentString = "No data returned";
-              } else if (
-                functionJsonResponse &&
-                typeof functionJsonResponse === "object"
-              ) {
-                contentString = chatItem.content;
               }
               return (
-                <DevChatItem
+                <UserChatItem
```

Comment:
why the change from Dev to User?

Reply:
This change is in `chat-ui`. The change enables developer mode to show a different viz to non-developer mode :smile: